### ISSUE TITLE
Make abstract class for entity without MissingPrimaryColumnError 

### ISSUE THREAD

**Issue by rubydoc** [typeorm/typeorm#1466]
Hi awesome team, 

I would like to make abstract class like this 
```
@Entity()
export abstract class CoreEntity extends BaseEntity {

    @BeforeInsert() 
    @BeforeUpdate() 
    async validateItem() {
        await validate(this).then(errors => {
            if (errors) {
                const error: ValidationError = errors[0];
                const message = _.values(error.constraints)[0];
                throw new ModelValidationError(message);
            } else {
                return true;
            }
        });  
    }
}
```

```
@Entity()
export class User extends CoreEntity {
}
```
Unfortunately, all tables doesn't have same primary key now. But I want to apply validator to all entities via inheritance. But when I implement like this, I've got MissingPrimaryColumnError. How can I ignore this error?

Thanks :)

**Comment by pleerock** [comment]
Having a primary columns is a requirements. There are lot of logic in ORM depend on primary columns and cannot be implemented without having them.

**Comment by rubydoc** [comment]
Thanks for the comment. I know the problem.
But, CoreEntity is not entity for real table. It is just ancestor object to apply validator to all entities. I want to ignore the warning. But I don't know where to handle the error.

**Comment by pleerock** [comment]
Its not necessary to mark your abstract entities with `@Entity()` decorator. Just remove it.

**Comment by rubydoc** [comment]
You are right, but I removed and tried, but same.

**Comment by pleerock** [comment]
Your concrete table still needs primary columns (User)

**Comment by rubydoc** [comment]
For example I have two entity User and Post, 
User has id as primary key. But Post has id_number as primary key.(I didn't agree this naming, but legacy system is working like this)
User and Post are inherited from CoreEntity. CoreEntity can't decide primary key name. But it raise the error. 

```
export abstract class CoreEntity extends BaseEntity {}
```
```
@Entity()
export class User extends CoreEntity {
    @PrimaryGeneratedColumn()
    id: number
}
```
```
@Entity()
export class Post extends CoreEntity {
    @PrimaryGeneratedColumn()
    id_number: number
}
```

I've got this error message
```
(node:52459) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): MissingPrimaryColumnError: Entity "CoreEntity" does not have a primary column. Primary column is required to have in all your entities. Use @PrimaryColumn decorator to add a primary column to your entity.
```

**Comment by rubydoc** [comment]
My file directory is
- entities
- core
and I set database configuration like this
```
const databaseConfig: MysqlConnectionOptions = {
    type: "mysql",
    port: 3306,
    host: (process.env.MYSQL_HOST || "localhost"),
    username: (process.env.MYSQL_USERNAME || "username"),
    password: (process.env.MYSQL_PASSWORD || "password"),
    database: process.env.MYSQL_DB,
    synchronize: false,
    logging: true,
    entities: [
        `${__dirname}/../entities/*.js`,
        `${__dirname}/../entities/*.ts`
    ]
};
```

**Comment by pleerock** [comment]
It should not rice errors, what typeorm version are you using? Can you try on latest and on `typeorm@next`? If issue still reproduce I'll need a PR with a failing test from your side or a minimal reproduction git repo.

**Comment by rubydoc** [comment]
Thanks @pleerock 
I solved the issued after updating version to 0.2.0-alpha.15(before 0.1.9) and removing dist folder. I realize to fix it when I creating sample project.

Thanks a lot for the kind answer
